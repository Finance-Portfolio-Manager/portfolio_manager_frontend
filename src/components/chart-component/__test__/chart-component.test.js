import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import ChartComponent from './../chart-component';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

import { shallow, ShallowWrapper, mount } from "enzyme";

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
const wrapper = shallow(<ChartComponent></ChartComponent>);

test("matches snapshot", ()=>{
    const tree = renderer.create(<ChartComponent></ChartComponent>).toJSON();
    expect(tree).toMatchSnapshot();
})

// test("useEffect hook", ()=>{
//     const mockResponseData = [{id: 1}, {id: 2}, {id: 3}];
//     const responseData = mockResponseData.map(e => ({...e}));
//     jest.clearAllMocks();
//     global.fetch = jest.fn(async () => ({
//       json: async () => mockResponseData
//     }));
//     const component = mount(<ChartComponent symbol="TSLA"></ChartComponent>);
// })