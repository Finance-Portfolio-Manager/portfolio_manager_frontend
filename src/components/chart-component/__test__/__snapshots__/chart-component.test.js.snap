// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="example"
>
  <div
    className="row"
  >
    <div
      className="mixed-chart"
    >
      <div
        height="auto"
        options={
          Object {
            "chart": Object {
              "height": 350,
            },
            "title": Object {
              "align": "left",
              "text": "Loading...",
            },
            "xaxis": Object {
              "convertedCatToNumeric": false,
              "type": "datetime",
            },
            "yaxis": Object {
              "tooltip": Object {
                "enabled": true,
              },
            },
          }
        }
        series={
          Array [
            Object {
              "data": Array [
                Array [
                  1990-01-02T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-03T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-04T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-05T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-06T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-07T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-08T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-09T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-10T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-11T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-12T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-13T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-14T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-15T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-16T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-17T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-18T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-19T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-20T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-21T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-22T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-23T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-24T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-25T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-26T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-27T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-28T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-29T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-30T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-01-31T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-01T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-02T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-03T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-04T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-05T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-06T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-07T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-08T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-09T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-10T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-11T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-12T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-13T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-14T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-15T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-16T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-17T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-18T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-19T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-20T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-21T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-22T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-23T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-24T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-25T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-26T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-27T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-02-28T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-01T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-02T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-03T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-04T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-05T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-06T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-07T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-08T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-09T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-10T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-11T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-12T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-13T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-14T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-15T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-16T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-17T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-18T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-19T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-20T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-21T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-22T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-23T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-24T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-25T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-26T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-27T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-28T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-29T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-30T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-03-31T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-04-01T00:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-04-01T23:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-04-02T23:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-04-03T23:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-04-04T23:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-04-05T23:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-04-06T23:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-04-07T23:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-04-08T23:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-04-09T23:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
                Array [
                  1990-04-10T23:00:00.000Z,
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ],
                ],
              ],
            },
          ]
        }
        type="line"
        width="100%"
      />
    </div>
  </div>
</div>
`;
